terracotta/server/flask_api.py

```diff
@@ -89,6 +80,19 @@ def create_app(debug: bool = False, profile: bool = False) -> Flask:
     import terracotta.server.singleband
     import terracotta.server.compute
 
+    new_app = Flask('terracotta.server')
+    new_app.debug = debug
+
+    # suppress implicit sort of JSON responses
+    new_app.config['JSON_SORT_KEYS'] = False
+
+    # CORS
+    settings = get_settings()
+    if settings.ALLOWED_ORIGINS_METADATA:
+        CORS(metadata_api, origins=settings.ALLOWED_ORIGINS_METADATA)
```

Comment:
It looks like this modifies the (global) blueprints in-place, so resetting the setting doesn't revert this. It might be cleanest to copy the global blueprints in this function if that's possible.

Reply:
On second thought, having multiple instances of Terracotta in the same Python thread is undefined behavior anyway. So you can also just set CORS unconditionally (with whatever is set in the settings, including no allowed origins). You might have to set the fixture scope for the test client to local in that case, so pytest creates a new one for every test.

Reply:
Maybe add a comment to the docs that the parameter has to be set before creating the app.

Reply:
> Maybe add a comment to the docs that the parameter has to be set before creating the app.

Oooh, yes, I did not think of that. Users must definitely be (made) aware that some config changes only go live when the TC server restarts, which in the case of Lambda happens when the functions goes cold or is updated, I guess.

Reply:
Since these Flask apps, blueprints and stuff have so much global state (and also our `terracotta.settings`), maybe we should ensure that tests run in separate processes with [`pytest-xdist` and `--forked`](https://stackoverflow.com/a/48310939/512111). 

I could make my tests pass by running disabled CORS first and enabled after, but that this works is just coincidental.

Reply:
Avoiding global state is nicer, so no need for xdist.